### PR TITLE
Fix up flawed log to log code

### PR DESCRIPTION
**Pull request by TheBunnyMan123**
I am currently not able to test this, so I will tomorrow, but if you can that would be great (if you are confident go ahead and merge it anyway)

### CODE REVIEW

**Comment by PoolloverNathan** [comment]
yeah my code reviews can be a bit harsh sometimes sorry

**Comment by TheBunnyMan123** [comment]
The jars produced by this will be what I use for the first modrinth release because I need to upload a version before it can work

**Comment by PoolloverNathan** [comment]
build fail? :(

**Comment by PoolloverNathan** [comment]
fucking semicolons.

**Comment by TheBunnyMan123** [comment]
semicolon

File: common/src/main/java/com/thekillerbunny/goofyplugin/lua/GoofyAPI.java [hunk 1]
```diff
@@ -30,6 +30,10 @@ public GoofyAPI(FiguraLuaRuntime runtime) {
         this.owner = runtime.owner;
     }
 
+    public void canLog() {
+        return ((Configs.LOG_OTHERS.value || FiguraMod.isLocal(owner.owner)) && owner.permissions.get(GoofyPermissionsPlugin.CAN_LOG) == 1)
```
Comment (PoolloverNathan) [inline common/src/main/java/com/thekillerbunny/goofyplugin/lua/GoofyAPI.java]: missed semicolon :(
```suggestion
        return ((Configs.LOG_OTHERS.value || FiguraMod.isLocal(owner.owner)) && owner.permissions.get(GoofyPermissionsPlugin.CAN_LOG) == 1);
```

File: common/src/main/java/com/thekillerbunny/goofyplugin/lua/GoofyAPI.java [hunk 2]
```diff
@@ -158,39 +162,39 @@ public void traceToLog(@LuaNotNil String data) {
         value = "goofy.is_debug_enabled"
     )
     public boolean isDebugEnabled() {
-        return !(!(owner.permissions.get(GoofyPermissionsPlugin.CAN_LOG) == 1) || (!Configs.LOG_OTHERS.value && !FiguraMod.isLocal(owner.owner)) || !GoofyPlugin.LOGGER.isDebugEnabled());
+        return !(!canLog() || !GoofyPlugin.LOGGER.isDebugEnabled());
```
Comment (PoolloverNathan) [inline common/src/main/java/com/thekillerbunny/goofyplugin/lua/GoofyAPI.java:165]: Morgan's law
```suggestion
        return canLog() && GoofyPlugin.LOGGER.isDebugEnabled();
```

File: common/src/main/java/com/thekillerbunny/goofyplugin/lua/GoofyAPI.java [hunk 3]
```diff
@@ -158,39 +162,39 @@ public void traceToLog(@LuaNotNil String data) {
...
     }
 
     @LuaWhitelist
     @LuaMethodDoc(
         value = "goofy.is_info_enabled"
     )
     public boolean isInfoEnabled() {
-        return !(!(owner.permissions.get(GoofyPermissionsPlugin.CAN_LOG) == 1) || (!Configs.LOG_OTHERS.value && !FiguraMod.isLocal(owner.owner)) || !GoofyPlugin.LOGGER.isInfoEnabled());
+        return !(!canLog() || !GoofyPlugin.LOGGER.isInfoEnabled());
```
Comment (PoolloverNathan) [inline common/src/main/java/com/thekillerbunny/goofyplugin/lua/GoofyAPI.java:173]: Morgan's law
```suggestion
        return canLog() && GoofyPlugin.LOGGER.isInfoEnabled();
```

File: common/src/main/java/com/thekillerbunny/goofyplugin/lua/GoofyAPI.java [hunk 4]
```diff
@@ -158,39 +162,39 @@ public void traceToLog(@LuaNotNil String data) {
...
     }
 
     @LuaWhitelist
     @LuaMethodDoc(
         value = "goofy.is_warn_enabled"
     )
     public boolean isWarnEnabled() {
-        return !(!(owner.permissions.get(GoofyPermissionsPlugin.CAN_LOG) == 1) || (!Configs.LOG_OTHERS.value && !FiguraMod.isLocal(owner.owner)) || !GoofyPlugin.LOGGER.isWarnEnabled());
+        return !(!canLog() || !GoofyPlugin.LOGGER.isWarnEnabled());
```
Comment (PoolloverNathan) [inline common/src/main/java/com/thekillerbunny/goofyplugin/lua/GoofyAPI.java:181]: Morgan's law
```suggestion
        return canLog() && GoofyPlugin.LOGGER.isWarnEnabled();
```

File: common/src/main/java/com/thekillerbunny/goofyplugin/lua/GoofyAPI.java [hunk 5]
```diff
@@ -158,39 +162,39 @@ public void traceToLog(@LuaNotNil String data) {
...
     }
 
     @LuaWhitelist
     @LuaMethodDoc(
         value = "goofy.is_error_enabled"
     )
     public boolean isErrorEnabled() {
-        return !(!(owner.permissions.get(GoofyPermissionsPlugin.CAN_LOG) == 1) || (!Configs.LOG_OTHERS.value && !FiguraMod.isLocal(owner.owner)) || !GoofyPlugin.LOGGER.isErrorEnabled());
+        return !(!canLog() || !GoofyPlugin.LOGGER.isErrorEnabled());
```
Comment (PoolloverNathan) [inline common/src/main/java/com/thekillerbunny/goofyplugin/lua/GoofyAPI.java:189]: Morgan's law
```suggestion
        return canLog() && GoofyPlugin.LOGGER.isErrorEnabled();
```

File: common/src/main/java/com/thekillerbunny/goofyplugin/lua/GoofyAPI.java [hunk 6]
```diff
@@ -158,39 +162,39 @@ public void traceToLog(@LuaNotNil String data) {
...
     }
 
     @LuaWhitelist
     @LuaMethodDoc(
         value = "goofy.is_trace_enabled"
     )
     public boolean isTraceEnabled() {
-        return !(!(owner.permissions.get(GoofyPermissionsPlugin.CAN_LOG) == 1) || (!Configs.LOG_OTHERS.value && !FiguraMod.isLocal(owner.owner)) || !GoofyPlugin.LOGGER.isTraceEnabled());
+        return !(!canLog() || !GoofyPlugin.LOGGER.isTraceEnabled());
```
Comment (PoolloverNathan) [inline common/src/main/java/com/thekillerbunny/goofyplugin/lua/GoofyAPI.java:197]: Morgan's law
```suggestion
        return canLog() && GoofyPlugin.LOGGER.isTraceEnabled();
```

File: common/src/main/java/com/thekillerbunny/goofyplugin/lua/GoofyAPI.java (hunk 5 above)
Comment (PoolloverNathan) [inline common/src/main/java/com/thekillerbunny/goofyplugin/lua/GoofyAPI.java:189]: Morgan's law
```suggestion
        return canLog() && GoofyPlugin.LOGGER.isErrorEnabled();
```

File: common/src/main/java/com/thekillerbunny/goofyplugin/lua/GoofyAPI.java (hunk 6 above)
Comment (PoolloverNathan) [inline common/src/main/java/com/thekillerbunny/goofyplugin/lua/GoofyAPI.java:197]: Morgan's law
```suggestion
        return canLog() && GoofyPlugin.LOGGER.isTraceEnabled();
```

**Comment by TheBunnyMan123** [comment]
return value types
I miss autocomplete

File: common/src/main/java/com/thekillerbunny/goofyplugin/lua/GoofyAPI.java [hunk 7]
```diff
@@ -30,6 +30,10 @@ public GoofyAPI(FiguraLuaRuntime runtime) {
         this.owner = runtime.owner;
     }
 
+    public void canLog() {
```
Comment (PoolloverNathan) [inline common/src/main/java/com/thekillerbunny/goofyplugin/lua/GoofyAPI.java]: there is a void in my soul only filled by booleans
```suggestion
    public boolean canLog() {
```

**Comment by TheBunnyMan123** [comment]
this is why we have a build check

**Review by PoolloverNathan — CHANGES_REQUESTED** [review]

**Comment by PoolloverNathan** [comment]
review is being sent over the 100kbps link please wait

File: common/src/main/java/com/thekillerbunny/goofyplugin/lua/GoofyAPI.java (hunk 7 above)
Comment (TheBunnyMan123) [reply common/src/main/java/com/thekillerbunny/goofyplugin/lua/GoofyAPI.java]: ahead of you

File: common/src/main/java/com/thekillerbunny/goofyplugin/lua/GoofyAPI.java (hunk 6 above)
Comment (TheBunnyMan123) [reply common/src/main/java/com/thekillerbunny/goofyplugin/lua/GoofyAPI.java:197]: implementing

**Comment by PoolloverNathan** [comment]
alright it made it

**Comment by PoolloverNathan** [comment]
morgan's law: `!(!a || !b) == a && b`

**Comment by TheBunnyMan123** [comment]
I'm doing this on a school computer so there I am doing this in the web browser

**Review by PoolloverNathan — APPROVED** [review]

**Comment by TheBunnyMan123** [comment]
build success, merging

**Pull request merged**